.github/workflows/build_and_run_chain_simulator_and_execute_system_test.yml

```diff
@@ -0,0 +1,341 @@
+          BRANCH_NAME: ${{ env.BRANCH_NAME }}
+          CURRENT_BRANCH: ${{ env.CURRENT_BRANCH }}
+          MX_CHAIN_GO_TARGET_BRANCH: ${{ env.MX_CHAIN_GO_TARGET_BRANCH }}
+          MX_CHAIN_SIMULATOR_TARGET_BRANCH: ${{ env.MX_CHAIN_SIMULATOR_TARGET_BRANCH }}
+          MX_CHAIN_TESTING_SUITE_TARGET_BRANCH: ${{ env.MX_CHAIN_TESTING_SUITE_TARGET_BRANCH }}
+          LATEST_COMMIT_HASH: ${{ env.LATEST_COMMIT_HASH }}
+          PYTEST_EXIT_CODE: ${{ env.PYTEST_EXIT_CODE }}
+
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const timestamp = process.env.TIMESTAMP;
+            const branchName = process.env.BRANCH_NAME;
+            const currentBranch = process.env.CURRENT_BRANCH;
+            const goTargetBranch = process.env.MX_CHAIN_GO_TARGET_BRANCH;
+            const simulatorTargetBranch = process.env.MX_CHAIN_SIMULATOR_TARGET_BRANCH;
+            const testingSuiteTargetBranch = process.env.MX_CHAIN_TESTING_SUITE_TARGET_BRANCH;
+            const commitHash = process.env.LATEST_COMMIT_HASH;
+            const exitCode = process.env.PYTEST_EXIT_CODE;
+            const issue_number = context.issue.number;
+            const owner = context.repo.owner;
+            const repo = context.repo.repo;
+            let message;
+
+            if (timestamp && branchName && timestamp !== "" && branchName !== "") {
+              const reportUrl = `https://multiversx.github.io/mx-chain-testing-suite/reports/${branchName}/${timestamp}/index.html`;
+              message = `
+              📊 **MultiversX Automated Test Report:** [View Report](${reportUrl})
+            
+              🔄 **Build Details:**
+              - **mx-chain-go Commit Hash:** \`${commitHash}\`
+              - **Current Branch:** \`${currentBranch}\`
+              - **mx-chain-go Target Branch:** \`${goTargetBranch}\`
+              - **mx-chain-simulator-go Target Branch:** \`${simulatorTargetBranch}\`
+              - **mx-chain-testing-suite Target Branch:** \`${testingSuiteTargetBranch}\`
+            
+              🚀 **Environment Variables:**
+            - **TIMESTAMP:** \`${timestamp}\`
+            - **PYTEST_EXIT_CODE:** \`${exitCode}\`
+              🎉 **MultiversX CI/CD Workflow Complete!**
+              `;
+            } else {
+              message = "⚠️ No report was generated due to an error or cancellation of the process.\nPlease checkout gh action logs for details";
+            }
+
+            github.rest.issues.createComment({
+              issue_number: issue_number,
+              owner: owner,
+              repo: repo,
+              body: message
+            });
+
+      - name: Fail job if tests failed
+        if: always()
+        run: |
+          if [ "${{ env.PYTEST_EXIT_CODE }}" != "0" ]; then
+            echo "Tests failed with exit code ${{ env.PYTEST_EXIT_CODE }}"
+            exit 1
+          else
+            echo "Tests passed successfully."
+          fi
```

Comment:
Just my 2 cents. I am leaving this here for posterity.

Don't get me wrong, I appreciate your effor into building this and believe it can offer great value to our organizaction, but the overall design is too complex and could be spread out into more parts. CI pipelines shouldn't be too complex in nature and should rely more on the binaries it invokes. Such as enviornment provisioning for example, one could more easily start a container than checking out a repository, building a binary with the correct go version and exposing it as an environment variable. This could be handled in the `mx-chain-testing-suite` repository for instance.

For instance `mx-chain-testing-suite` already has a job that triggers the tests. We can trigger that job from this respective workflow through a webhook and wait for its response. In the webhook's body you can put the dependencies required to build the `chain-simulator` binary needed for the tests. It would be easier with containers, but that requires a bit extra overhead, such as a private container registry, but from my knowledge we do have something that could be used @liviuf-ancas @steliandoruzaharia.

Reply:
It's good to have a containerized testing framework, and we can consider that for future improvements. For now, let's focus on the current implementation and at least start with what we have.